### PR TITLE
Feature/708 stored query versioning

### PR DESCRIPTION
**Pull request by HolgerReiseVSys**
# Changes

Improved versioning of stored queries

# Related issue

> Reference related issues, and use one of the [closing keywords, e.g. closes or fixes](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link the corresponding issue, if any.

# Additional information 

> Provide additional information for this change, if needed.

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: service/src/main/java/org/ehrbase/dao/access/util/VersionConflictException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehrbase.dao.access.util;
+
+public class VersionConflictException extends RuntimeException {
```
Comment (dmitrytulba) [inline service/src/main/java/org/ehrbase/dao/access/util/VersionConflictException.java:20]: this class is never used VersionConflictException

File: service/src/main/java/org/ehrbase/service/QueryServiceImp.java [hunk 2]
```diff
@@ -108,7 +112,7 @@ public QueryResultDto query(
                         auditResultMap);
 
             default:
-                throw new IllegalArgumentException("Invalid query mode:" + queryMode);
+                throw new IllegalArgumentException("Invalid query mode: " + queryMode);
```
Comment (dmitrytulba) [inline service/src/main/java/org/ehrbase/service/QueryServiceImp.java:115]: It good to replace all concatenation strings to ~ String.format

**Review by dmitrytulba — APPROVED** [review]
LGTM

File: service/src/main/java/org/ehrbase/dao/access/jooq/StoredQueryAccess.java [hunk 3]
```diff
@@ -46,134 +57,149 @@ public StoredQueryAccess(I_DomainAccess domainAccess, StoredQueryRecord queryRec
...
 
-        List<StoredQueryRecord> queryRecords;
+            if (semVer.isNoVersion()) {
+                sb.append("\\d+");
+            } else {
+                sb.append(semVer.toVersionString().replace(".", "\\."));
+            }
 
-        // TODO: retrieve a stored query using a partial SEMVER
+            Stream.of(semVer.minor(), semVer.patch())
```
[the remark this answers is not in the document]
Comment (HolgerReiseVSys) [reply service/src/main/java/org/ehrbase/dao/access/jooq/StoredQueryAccess.java:124]: This also allows for unit testing the regex

**Review by dmitrytulba — APPROVED** [review]

**Pull request merged**